sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze.py

```diff
@@ -0,0 +1,1313 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+import os
+import pytest
+import platform
+import functools
+
+from azure.core.exceptions import HttpResponseError, ClientAuthenticationError
+from azure.core.credentials import AzureKeyCredential
+from testcase import TextAnalyticsTest, GlobalTextAnalyticsAccountPreparer
+from testcase import TextAnalyticsClientPreparer as _TextAnalyticsClientPreparer
+from azure.ai.textanalytics import (
+    TextAnalyticsClient,
+    EntitiesRecognitionTask,
+    PiiEntitiesRecognitionTask,
+    KeyPhraseExtractionTask,
+    TextDocumentInput,
+    VERSION,
+    TextAnalyticsApiVersion,
+)
+
+# pre-apply the client_cls positional argument so it needn't be explicitly passed below
+TextAnalyticsClientPreparer = functools.partial(_TextAnalyticsClientPreparer, TextAnalyticsClient)
+
+class TestAnalyze(TextAnalyticsTest):
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": os.environ.get('AZURE_TEXT_ANALYTICS_ENDPOINT')
+    })
+    def test_no_single_input(self, client):
+        with self.assertRaises(TypeError):
+            response = client.begin_analyze("hello world")
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": os.environ.get('AZURE_TEXT_ANALYTICS_ENDPOINT')
+    })
+    def test_all_successful_passing_dict_key_phrase_task(self, client):
+        docs = [{"id": "1", "language": "en", "text": "Microsoft was founded by Bill Gates and Paul Allen"},
+                {"id": "2", "language": "es", "text": "Microsoft fue fundado por Bill Gates y Paul Allen"}]
+
+        response = client.begin_analyze(
+            docs, 
+            key_phrase_extraction_tasks=[KeyPhraseExtractionTask()], 
+            show_stats=True
+        ).result()
+
+        results_pages = list(response)
+        self.assertEqual(len(results_pages), 1)
+
+        task_results = results_pages[0].key_phrase_extraction_results
+        self.assertEqual(len(task_results), 1)
+
+        results = task_results[0].results
+        self.assertEqual(len(results), 2)
+
+        for phrases in results:
+            self.assertIn("Paul Allen", phrases.key_phrases)
+            self.assertIn("Bill Gates", phrases.key_phrases)
+            self.assertIn("Microsoft", phrases.key_phrases)
+            self.assertIsNotNone(phrases.id)
+            # self.assertIsNotNone(phrases.statistics)
```

Comment:
curious why this is commented?

Reply:
I commented out a lot of the checks for statistics since the service isn't currently supporting this in the result just yet.  The plan is for it to be supported soon, but probably won't be fixed before the first release.  I chose to leave it here for now until it can be implemented later but if you think I should remove them let me know.